components/ChallengeMaterial.tsx

```diff
@@ -82,23 +94,51 @@ export const ChallengeQuestionCard: React.FC<ChallengeQuestionCardProps> = ({
   )
 }
 
-const ChallengeMaterial: React.FC<ChallengeMaterialProps> = props => {
-  if (!props.challenges) {
+const ChallengeMaterial: React.FC<ChallengeMaterialProps> = ({
+  userSubmissions,
+  challenges
+}) => {
+  if (!challenges.length) {
     return <h1>No Challenges for this lesson</h1>
   }
-  const sortedChallenges: Challenge[] = props.challenges.sort(
+  //create an object to evaluate the student's status with a challenge
+  const userSubmissionsObject = userSubmissions.reduce(
+    (acc: UserSubmissionsObject, submission: UserSubmission) => {
+      acc[submission.challengeId] = submission
+      return acc
+    },
+    {}
+  )
+  //create a new Challenges array with user submission data integrated
+  const challengesWithSubmissionData = challenges.reduce(
+    (acc: ChallengeSubmissionData[], challenge: Challenge) => {
+      const submissionData = userSubmissionsObject[challenge.id] || {}
+      acc.push(
+        Object.assign(challenge, {
```

Comment:
Object.assign should be replaced with `{...challenge, status..., submission}`. Could we use `submission` instead of `submissionData`?